backends/tfhe-cuda-backend/cuda/src/crypto/fast_packing_keyswitch.cuh

```diff
@@ -0,0 +1,350 @@
+#ifndef CNCRT_FAST_KS_CUH
+#define CNCRT_FAST_KS_CUH
+
+#undef NDEBUG
+#include <assert.h>
+
+#include "device.h"
+#include "gadget.cuh"
+#include "helper_multi_gpu.h"
+#include "keyswitch.cuh"
+#include "polynomial/functions.cuh"
+#include "polynomial/polynomial_math.cuh"
+#include "torus.cuh"
+#include "utils/helper.cuh"
+#include "utils/kernel_dimensions.cuh"
+#include <thread>
+#include <vector>
+
+#define CEIL_DIV(M, N) ((M) + (N)-1) / (N)
+
+const int BLOCK_SIZE_GEMM = 64;
+const int THREADS_GEMM = 8;
+const int BLOCK_SIZE_DECOMP = 8;
+
+__host__ inline bool can_use_pks_fast_path(uint32_t lwe_dimension_in,
+                                           uint32_t num_lwe,
+                                           uint32_t polynomial_size,
+                                           uint32_t level_count,
+                                           uint32_t glwe_dimension) {
+  // TODO: Generalize to level_count > 1 by transposing the KSK
+  return level_count == 1;
+}
+
+template <typename Torus, typename TorusVec>
+__global__ void decompose_vectorize_init(Torus const *lwe_in, Torus *lwe_out,
+                                         uint32_t lwe_dimension_in,
```

Comment:
this was the name in `packing_keyswitch_lwe_ciphertext_into_glwe_ciphertext`. It seems to be because packing KS was derived from reglular KS which had a `lwe_dimension_out`. for packing we don't have a `lwe_dimension_out` so I'll change this as you suggest.